### DISCUSSION TITLE
Creating a grid of buttons using button_html parameter

### DISCUSSION THREAD

**Discussion by yuvalharr** [jspsych/jsPsych#780] [Q&A]
Hello,

I'm struggling with arranging a grid of buttons in the html-button-response plugin.
What I'm trying to do is to show a 3X3 grid of buttons, each button a part of a whole image, like shown here:
![image](https://user-images.githubusercontent.com/40396020/85554150-4ffd3800-b62d-11ea-92a3-2a893f13c774.png)

Each trial the timeline passes an array of 9 images paths ('grid_images'), and I'm trying to customize the button_html parameter so that it will show them arranged in a grid. This is the trial that is being passed:
```
{
         type: 'html-button-response',
         stimulus: [],
         choices: jsPsych.timelineVariable('grid_images'),
         button_html: '<div class="wrapper"><div><img src=%choice%></div></div>',
         prompt: '<p style="font-size: 30px;">Where is the item located?</p>'
}
```

And this is the 'wrapper' class that I made inside the css file to try manipulate the images into a grid:
```
.wrapper {
  display: grid;
  grid-template-columns: auto auto auto;
  grid-gap: 10px;
}
```

For now it seems to me that they are not arranged into a grid but rather each one of them is a one item grid of its own. Every zoom in and out changes the way the images are arranged. for example, this is a 25% zoom view:
  
![image](https://user-images.githubusercontent.com/40396020/85555288-7079c200-b62e-11ea-8e51-7d819758ee36.png)

Any help would be greatly appreciated!

**Comment by vijaymarupudi** [comment]
I personally don't think the `html-button-response` plugin is the right tool for the job here. CSS grid typically requires a dom structure that's not customizable by the plugin. I would consider making a new plugin to complete this task.

~ Vijay

**Comment by jenniekim00** [comment]


**Reply by yuvalharr** [reply]
Thanks a lot, I'll have a look!

**Reply by conbainbridge** [reply]
Agreed that combining html-button-response with vsl-grid-scene would be a great starting point for customizing a plugin! One approach you could take is:

1. Start from the html-button-response (so data saving is already set up for you)
2. Copy over the table setup code from vsl-grid-scene, using the button variable (which is built off of the choices parameter of the plugin) to set up the rows and columns
3. Add the image_size parameter from vsl-grid-scene to the custom button plugin info to set grid cell size.
4. Wrap the str variable that gets built into the button in an img tag.

Hope this helps!
